src/test/java/lingogo/logic/commands/TestCommandTest.java

```diff
@@ -0,0 +1,138 @@
+package lingogo.logic.commands;
+
+import static lingogo.logic.commands.CommandTestUtil.VALID_ENGLISH_PHRASE_AFTERNOON;
+import static lingogo.logic.commands.CommandTestUtil.VALID_ENGLISH_PHRASE_GOOD_MORNING;
+import static lingogo.logic.commands.CommandTestUtil.assertCommandFailure;
+import static lingogo.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static lingogo.logic.commands.CommandTestUtil.showFlashcardAtIndex;
+import static lingogo.testutil.TypicalFlashcards.getTypicalFlashcardApp;
+import static lingogo.testutil.TypicalIndexes.INDEX_FIRST_FLASHCARD;
+import static lingogo.testutil.TypicalIndexes.INDEX_SECOND_FLASHCARD;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+import lingogo.commons.core.Messages;
+import lingogo.commons.core.index.Index;
+import lingogo.model.Model;
+import lingogo.model.ModelManager;
+import lingogo.model.UserPrefs;
+import lingogo.model.flashcard.Flashcard;
+import lingogo.model.flashcard.Phrase;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code TestCommand}.
+ */
+public class TestCommandTest {
+
+    private Model model = new ModelManager(getTypicalFlashcardApp(), new UserPrefs());
+    private Phrase validPhraseAfternoon = new Phrase(VALID_ENGLISH_PHRASE_AFTERNOON);
+    private Phrase validPhraseGoodMorning = new Phrase(VALID_ENGLISH_PHRASE_GOOD_MORNING);
+
+    @Test
+    public void execute_validIndexUnfilteredListCorrectGivenPhrase_success() {
+        Flashcard flashcardToTest = model.getFilteredFlashcardList().get(INDEX_FIRST_FLASHCARD.getZeroBased());
+        TestCommand testCommand = new TestCommand(INDEX_FIRST_FLASHCARD, validPhraseAfternoon);
+
+        String expectedMessage = String.format(testCommand.MESSAGE_TEST_FLASHCARD_SUCCESS_CORRECT,
+            flashcardToTest.getForeignPhrase(), flashcardToTest.getEnglishPhrase(), VALID_ENGLISH_PHRASE_AFTERNOON);
+
+        ModelManager expectedModel = new ModelManager(model.getFlashcardApp(), new UserPrefs());
+
+        assertCommandSuccess(testCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_validIndexUnfilteredListWrongGivenPhrase_success() {
+        Flashcard flashcardToTest = model.getFilteredFlashcardList().get(INDEX_FIRST_FLASHCARD.getZeroBased());
+        TestCommand testCommand = new TestCommand(INDEX_FIRST_FLASHCARD, validPhraseGoodMorning);
+
+        String expectedMessage = String.format(testCommand.MESSAGE_TEST_FLASHCARD_SUCCESS_WRONG,
```

Comment:
```suggestion
        String expectedMessage = String.format(TestCommand.MESSAGE_TEST_FLASHCARD_SUCCESS_WRONG,
```
Would be good practice to standardize and use static variables and via the class instead of an instance.